st2common/st2common/operators.py

```diff
@@ -114,19 +113,33 @@ def greater_than(value, criteria_pattern):
     return value > criteria_pattern
 
 
-def match_wildcard(value, criteria_pattern):
+def match_regex(value, criteria_pattern):
     if criteria_pattern is None:
         return False
+    regex = re.compile(criteria_pattern)
+    # check for a match and not for details of the match.
+    return regex.match(value) is not None
 
-    return fnmatch.fnmatch(value, criteria_pattern)
+def imatch_regex(value, criteria_pattern):
+    if criteria_pattern is None:
+        return False
+    regex = re.compile(criteria_pattern, re.IGNORECASE)
+    # check for a match and not for details of the match.
+    return regex.match(value) is not None
 
+def search_regex(value, criteria_pattern):
+    if criteria_pattern is None:
+        return False
+    regex = re.compile(criteria_pattern)
+    # check for a match and not for details of the match.
+    return regex.search(value) is not None
 
-def match_regex(value, criteria_pattern):
+def isearch_regex(value, criteria_pattern):
     if criteria_pattern is None:
         return False
-    regex = re.compile(criteria_pattern, re.DOTALL)
```

Comment:
I think DOTALL was added to allow regex to work with \n in the string IIRC. Please don't remove DOTALL.